napari/components/viewer_model.py

```diff
@@ -658,6 +658,23 @@ def add_image(
 
             return layer_list
 
+    def load_sample_data(self, plugin_name: str, sample_name: str, **kwargs):
+        from ..plugins import _sample_data
+
+        try:
+            data = _sample_data[plugin_name][sample_name]
+        except KeyError:
+            raise KeyError(
+                f"Plugin {plugin_name!r} does not provide sample data "
+                f"named {sample_name!r}."
+            )
+
+        if callable(data):
+            for datum in data(**kwargs):
```

Comment:
this would actually let the data-getting function accept parameters (#1211).  imagine, for instance, the shepp-logan phantom sample in imageJ, that lets you give dimensions... or perhaps a PSF generator.  The visual representation of that would still need some work/discussion, but it would let you do `viewer.open_sample('plugin', 'sample', nx=512, dz=0.13)` ... or something.

Reply:
actually, a good example of this in the scikit image plugin is:
`viewer.open_sample('scikit-image', 'binary_blobs', length=128, n_dim=3, volume_fraction=0.3)`

Reply:
yes, that was the one that came to my mind after your comment! I use that one all the time!!